client/client.go

```diff
@@ -102,7 +109,38 @@ func NewApiClient(conf Config) (*ApiClient, error) {
 	return New(conf)
 }
 
-// Client initialization
+// keepaliveDialer is used to set keepalive options on the socket used for
+// connection. This is for handling connection timeouts while server
+// is not accessible after connection was alrady started.
+type keepaliveDialer struct {
+	net.Dialer
+}
```

Comment:
Looks like `net.Dialer` already [has a field](https://golang.org/pkg/net/#Dialer) `KeepAlive`, could you just use that instead of the `tcpkeepalive` module?